crates/provider/src/builder.rs

```diff
@@ -73,25 +107,65 @@ where
 ///
 /// [`tower::ServiceBuilder`]: https://docs.rs/tower/latest/tower/struct.ServiceBuilder.html
 #[derive(Debug)]
-pub struct ProviderBuilder<L, N = Ethereum> {
+pub struct ProviderBuilder<L, F, N = Ethereum> {
     layer: L,
-    network: PhantomData<N>,
+    filler: F,
+    network: PhantomData<fn() -> N>,
 }
 
-impl ProviderBuilder<Identity, Ethereum> {
+impl ProviderBuilder<Identity, Identity, Ethereum> {
     /// Create a new [`ProviderBuilder`].
     pub const fn new() -> Self {
-        ProviderBuilder { layer: Identity, network: PhantomData }
+        ProviderBuilder { layer: Identity, filler: Identity, network: PhantomData }
     }
 }
 
-impl<N> Default for ProviderBuilder<Identity, N> {
+impl<N> Default for ProviderBuilder<Identity, Identity, N> {
     fn default() -> Self {
-        ProviderBuilder { layer: Identity, network: PhantomData }
+        ProviderBuilder { layer: Identity, filler: Identity, network: PhantomData }
+    }
+}
+
+impl<L, N> ProviderBuilder<L, Identity, N> {
+    /// Add preconfigured set of layers handling gas estimation and nonce management
+    pub fn with_recommended_fillers(self) -> ProviderBuilder<L, RecommendFiller, N> {
+        self.filler(GasFiller).filler(NonceFiller::default()).filler(ChainIdFiller::default())
+    }
```

Comment:
the answer is no*. no because `with_chain_id` is implemented only when the builder has the `Identity` nop filler. 

* you could do `with_recommended_fillers().filler(ChainIdFiller::new(Some(1)))` but then you would have multiple `ChainId` fillers